### PR TITLE
Re-Brand To Bitcoin Classic

### PR DESCRIPTION
**Pull request by ahmedbodi**
Rebrand code to Bitcoin Classic. 
Updated Dates to Bitcoin Core, Kept in their Copyright Line
and added ours as 2015-2016.

Standardised (almost) the Copyright lines to be consistent in all files rather than different ones in each

### CODE REVIEW

**Comment by gavinandresen** [comment]
ACK on branding.

NAK on changing copyrights on every file; copyright should only be touched as files are edited, and gratuitous changes are a bad idea. Part of the reason I got fed up with working on core was constant rebasing due to trivial whitespace (or copyright or refactor or whatever) changes.

**Comment by ahmedbodi** [comment]
Hmm, Okay, was not aware of that one would've made my life a lot easier haha. Will fix it now

**Comment by ahmedbodi** [comment]
@gavinandresen reverted and a new version without the copyrights committed. command used to determine all instances of Core which were not comments is:
grep "Bitcoin Core" . -r | grep "//" --invert | grep "# Copyright" --invert

**Comment by FinalHash** [comment]
This looks good. Will merge tomorrow. @jtoomim or @ahmedbodi or @gavinandresen have you built it yet? Ill try if i can get some time tonight

**Comment by gavinandresen** [comment]
Compiles and runs fine on my OSX machine.

The "Revert" commits should be squashed from revision history before merge.

**Comment by FinalHash** [comment]
Good. I'll push the history down and merge in the AM
On Jan 8, 2016 16:28, Gavin Andresen notifications@github.com wrote:Compiles and runs fine on my OSX machine.

The "Revert" commits should be squashed from revision history before merge.

File: src/qt/bitcoinstrings.cpp [hunk 1]
```diff
@@ -9,111 +9,111 @@
...
 "Set maximum size of high-priority/low-fee transactions in bytes (default: %d)"),
-QT_TRANSLATE_NOOP("bitcoin-core", ""
+QT_TRANSLATE_NOOP("bitcoin-classic", ""
 "Set the number of script verification threads (%u to %d, 0 = auto, <0 = "
-"leave that many cores free, default: %d)"),
-QT_TRANSLATE_NOOP("bitcoin-core", ""
-"Set the number of threads for coin generation if enabled (-1 = all cores, "
+"leave that many classics free, default: %d)"),
+QT_TRANSLATE_NOOP("bitcoin-classic", ""
+"Set the number of threads for coin generation if enabled (-1 = all classics, "
```
Comment (dgmib) [inline src/qt/bitcoinstrings.cpp]: This line shouldn't change.

**Comment by dgmib** [comment]
Compiles and runs for me on Ubuntu

**Comment by jtoomim** [comment]
I've made a consider.it page for this PR here: https://bitcoinclassic.consider.it/merge-2-re-brand-to-bitcoin-classic. If you think it's ready to be merged as-is, you can vote in favor. If you have looked at it and have seen issues, or if you think people are voting for it without having given everybody sufficient time for code review, you can vote against it. Keep in mind you can always change your vote later, so don't be afraid to vote against something that has fixable issues.

**Comment by MathieuGorichon** [comment]
The user agent should be modified to "Classic". I guess it is in src/clientversion.cpp.

**Comment by jtoomim** [comment]
@ahmedbodi, I've addressed the two issues raised so far ("Cores free", user agent string) in a PR to your repo/branch. https://github.com/Multicoin-co/bitcoinclassic/pull/1

**Comment by jtoomim** [comment]
https://bitcoinclassic.consider.it/merge-2-re-brand-to-bitcoin-classic shows pretty strong support for this, and no opposition, and the only criticisms that have been raised so far have been addressed. 

Since the code was changed once since the original tested ACKs, I'd like one more tested re-ACK before merging.

**Comment by ahmedbodi** [comment]
Once the commits are squashed into one its an ACK from me

**Comment by chriswheeler** [comment]
There are a number of URLs which are pointing to bitcoin.org and third party service accounts related to Bitcoin Core (Travis CI status indicator, transifex, github and maybe more). 

Should they be fixed as part of this PR or in another?

Also, the Github description/project link needs updating.

**Comment by ahmedbodi** [comment]
I'd rather they be updated in seperate ones to be honest. This one was supposed to just be for renaming. thats why i left them out. We dont yet have a travis-ci or transifex setup so thats one of the reasons they havent been changed yet

File: README.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ Translations
 ------------
 
 Changes to translations as well as new translations can be submitted to
-[Bitcoin Core's Transifex page](https://www.transifex.com/projects/p/bitcoin/).
+[Bitcoin Classic's Transifex page](https://www.transifex.com/projects/p/bitcoin/).
```
Comment (tobixen) [inline README.md:77]: The URL may need changing as well

**Comment by tobixen** [comment]
I suppose this work has already been done with Bitcoin Unlimited and BitcoinXT.  Does the changeset compare?  (could we have saved time here by just taking one of the existing changesets, done a s/XT/ Classic/ and patched Bitcoin Core with the new changeset?)

**Comment by ahmedbodi** [comment]
I dont believe they do. XT did this but they also had other commits in their base which we dont approve off. Ive not seen unlimited so cannot compare there. We all felt it better to fork off Core rather than anything else

**Comment by jtoomim** [comment]
I agree with @ahmedbodi on having the URLs be done separately. 

I agree with @tobixen that looking at other rebranding jobs would be a good idea. Maybe @tobixen has enough time to look for them? Maybe just compare the files changed and linecounts changed to look for discrepancies?

**Comment by tobixen** [comment]
https://github.com/bitcoinxt/bitcoinxt/commit/e1816f4bda6a79f679fb5f87b324b542891aa25b for XT.  Can't compare the number of lines directly, as there are also bigger documentation changes there, but I will check some few things.

File: contrib/gitian-descriptors/gitian-osx-signer.yml [hunk 3]
```diff
@@ -34,5 +34,5 @@ script: |
 
   tar -xf ${UNSIGNED}
```
Comment (tobixen) [inline contrib/gitian-descriptors/gitian-osx-signer.yml:35]: Here is a changeset from bitcoinxt, further up in this file:

-- "url": "https://github.com/bitcoin/bitcoin-detached-sigs.git"
+- "url": "https://github.com/bitcoinxt/bitcoin-detached-sigs.git"

**Comment by tobixen** [comment]
I'd like to pull the branch locally or download the full diff, but I can't figure out the URL to do so?

I overlooked all instances of s/Core/XT/ and looked deeper into the remaining changes ...

There are some references to Bitcoin-Qt in the XT diff, have those been renamed properly in this PR?

There is an URL reference to www.bitcoin.org that have been changed.  

Some integers have been changed, I assume it's for the logo icon size.

CLIENT_NAME("Satoshi"); has been replaced with CLIENT_NAME("BitcoinXT") in clientversion.cpp.

"Bitcoin version" has been changed to Bitcoin XT version".

That's pretty much it.

**Comment by tobixen** [comment]
I've briefly looked through unlimited and commit 55b2c7119d681c6725e7edad33c8964180fcf0dc.  Seems to be a clean s/XT/UL/, plus that the project URL was changed.

**Comment by jtoomim** [comment]
@tobixen https://github.com/Multicoin-co/bitcoinclassic/tree/rebrand_bigblocks

File: README.md (hunk 2 above)
Comment (sugruedes) [inline README.md:77]: I imagine we need to create discrete accounts on transfex for classic?

File: contrib/debian/copyright [hunk 4]
```diff
@@ -5,9 +5,9 @@ Upstream-Contact: Satoshi Nakamoto <satoshin@gmx.com>
 Source: https://github.com/bitcoin/bitcoin
 
 Files: *
-Copyright: 2009-2012, Bitcoin Core Developers
+Copyright: 2009-2012, Bitcoin Classic Developers
 License: Expat
-Comment: The Bitcoin Core Developers encompasses the current developers listed on bitcoin.org,
+Comment: The Bitcoin Classic Developers encompasses the current developers listed on bitcoin.org,
```
Comment (bgok) [inline contrib/debian/copyright]: Remove reference to bitcoin.org

File: doc/release-notes.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-Bitcoin Core version 0.11.2 is now available from:
+Bitcoin Classic version 0.11.2 is now available from:
 
   <https://bitcoin.org/bin/bitcoin-core-0.11.2/>
```
Comment (bgok) [inline doc/release-notes.md:3]: This is a link to the core package

File: share/qt/Info.plist.in [hunk 6]
```diff
@@ -17,7 +17,7 @@
   <string>APPL</string>
 
   <key>CFBundleGetInfoString</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@, Copyright © 2009-@COPYRIGHT_YEAR@ The Bitcoin Core developers</string>
+  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@, Copyright © 2009-@COPYRIGHT_YEAR@ The Bitcoin lassic developers</string>
```
Comment (bgok) [inline share/qt/Info.plist.in]: typo

**Comment by jojva** [comment]
I did the job of rebranding Bitcoin Classic on my own and the result is very similar to what you guys did. I submitted a PR to Ahmed for the few things I noticed here: Multicoin-co/bitcoinclassic#3.

File: doc/README.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
-Bitcoin Core 0.11.2
+Bitcoin Classic 0.11.2
 =====================
 
 Setup
 ---------------------
-[Bitcoin Core](http://bitcoin.org/en/download) is the original Bitcoin client and it builds the backbone of the network. However, it downloads and stores the entire history of Bitcoin transactions (which is currently several GBs); depending on the speed of your computer and network connection, the synchronization process can take anywhere from a few hours to a day or more.
+[Bitcoin Classic](http://bitcoin.org/en/download) is the original Bitcoin client and it builds the backbone of the network. However, it downloads and stores the entire history of Bitcoin transactions (which is currently several GBs); depending on the speed of your computer and network connection, the synchronization process can take anywhere from a few hours to a day or more.
```
Comment (michaelgpearce) [inline doc/README.md]: Needs different copy here

File: doc/README_windows.txt [hunk 8]
```diff
@@ -13,7 +13,7 @@ Setup
 -----
 Unpack the files into a directory and run bitcoin-qt.exe.
 
-Bitcoin Core is the original Bitcoin client and it builds the backbone of the network.
+Bitcoin Classic is the original Bitcoin client and it builds the backbone of the network.
```
Comment (michaelgpearce) [inline doc/README_windows.txt]: Needs different copy here

File: contrib/debian/copyright (hunk 4 above)
Comment (ahmedbodi) [inline contrib/debian/copyright]: I dont believe we have a full site yet, once we do these will be changed. i requested that all url's be changed in a seperate PR

**Comment by dskloet** [comment]
The diff base is too big to be displayed by github. I would like to suggest making multiple smaller changes so they are easier to review.

Especially with moles trying to sneak bugs into the code it's important to be able to review changes carefully.

**Comment by jojva** [comment]
Use git. The diff may be big, but it's not complex. It's basically `s/Bitcoin Core/Bitcoin Classic`, there's not much to be afraid of.

**Comment by dskloet** [comment]
Forcing people to use a certain tool results in fewer people reviewing.
Smaller changes can be reviewed by different people while still guaranteeing that all changes are reviewed by some people.
Especially with changes where you think there's not much to be afraid of, it's easy to skim and miss something, either unintentional or malicious.

**Comment by jojva** [comment]
How do you do a review on github? How do you run the code on github?
I have built and run @ahmedbodi's branch on my computer and checked the full diff on git.
If you're skilled enough to review a patch, you're probably skilled enough to clone the repository and check the diff.
Splitting this patch would be wasting developer's time, but if you're willing to do so please go for it, we'll be glad to have one more on board. :)

**Comment by dskloet** [comment]
Doesn't the continuous build build and run the tests?
Being skilled enough to clone a repository doesn't mean everybody will do it.
If you want contributions, you should be welcoming to contributors by making it easy.
Making smaller changes doesn't take much time.

File: contrib/macdeploy/macdeployqtplus [hunk 9]
```diff
@@ -757,7 +757,7 @@ if config.dmg is not None:
     
     if fancy is None:
         try:
-            runHDIUtil("create", dmg_name, srcfolder="dist", format="UDBZ", volname="Bitcoin-Core", ov=True)
+            runHDIUtil("create", dmg_name, srcfolder="dist", format="UDBZ", volname="Bitcoin-Classic", ov=True)
```
Comment (dskloet) [inline contrib/macdeploy/macdeployqtplus:760]: Define a constant for the name and use it throughout the file.

Comment (jojva) [inline contrib/macdeploy/macdeployqtplus:760]: Indeed, that should be the next step after this gets merged.

**Comment by dskloet** [comment]
Many files become inconsistent with this change by saying "Bitcoin Classic" but then pointing to a URL that belongs to Core. Doing smaller chunks would also allow files to at least remain consistent with themselves.

**Comment by gavinandresen** [comment]
Changes in src/qt/locale will be overwritten when translations are pulled in from Transifex, they should not be changed with this pull.

I don't know if Transifex has a global across-translations search-and-replace function....

File: src/qt/bitcoinstrings.cpp [hunk 10]
```diff
@@ -9,111 +9,111 @@
...
 "notation for IPv6"),
-QT_TRANSLATE_NOOP("bitcoin-core", ""
+QT_TRANSLATE_NOOP("bitcoin-classic", ""
 "Bind to given address to listen for JSON-RPC connections. Use [host]:port "
 "notation for IPv6. This option can be specified multiple times (default: "
 "bind to all interfaces)"),
-QT_TRANSLATE_NOOP("bitcoin-core", ""
-"Cannot obtain a lock on data directory %s. Bitcoin Core is probably already "
+QT_TRANSLATE_NOOP("bitcoin-classic", ""
+"Cannot obtain a lock on data directory %s. Bitcoin Classic is probably already "
```
Comment (dskloet) [inline src/qt/bitcoinstrings.cpp:33]: Does Classic use a different data directory lock or would it also be locked if Bitcoin Core is already running?
Should it say "A Bitcoin client is probably already running."?

**Comment by jojva** [comment]
Well, I guess the diff is going to be severely shrinked. Happy @dskloet ? ^^

File: src/init.cpp [hunk 11]
```diff
@@ -1267,10 +1267,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
                 InitWarning(msg);
             }
             else if (nLoadWalletRet == DB_TOO_NEW)
-                strErrors << _("Error loading wallet.dat: Wallet requires newer version of Bitcoin Core") << "\n";
+                strErrors << _("Error loading wallet.dat: Wallet requires newer version of Bitcoin Classic") << "\n";
```
Comment (dskloet) [inline src/init.cpp:1270]: Or a new version of Core would do as well?

File: doc/translation_process.md [hunk 12]
```diff
@@ -106,6 +106,6 @@ To create a new language template, you will need to edit the languages manifest
 **Note:** that the language translation file **must end in `.qm`** (the compiled extension), and not `.ts`.
 
 ### Questions and general assistance
-The Bitcoin-Core translation maintainers include *tcatm, seone, Diapolo, wumpus and luke-jr*.You can find them, and others, in the Freenode IRC chatroom - `irc.freenode.net #bitcoin-dev`.
+The Bitcoin Classic translation maintainers include *tcatm, seone, Diapolo, wumpus and luke-jr*.You can find them, and others, in the Freenode IRC chatroom - `irc.freenode.net #bitcoin-dev`.
```
Comment (dskloet) [inline doc/translation_process.md:109]: I don't think those people are maintainers for anything related to Classic.

**Comment by christophebiocca** [comment]
Gavin's deletion of the 0.11.2 (XT variant) branch auto-closed this PR. Feel free to reopen after cherry picking it. As mentioned, please leave out the transifex stuff.

**Pull request closed without merging**